.designs/observables.md

```diff
@@ -0,0 +1,109 @@
+## Using Observables for subscription
+
+current implementation (Subscription Manager) is being used for a while now, and there are quiet few problems with it, which i want to address and show they are being solved using observables as the return value of the subscription resolver instead of current approach.
+
+# Data source limitation
+Current apollo implementation requires setting up a new WS server and provide it with a subscription protocol manager (subscription-transport-ws object) that defines the message types and handlers. 
+
+The protocol manager needs a subscription manager object in order to handle the actual subscribe, unsubscribe and publish events.
+
+The subscription manager class transfers the responsibility of storing and handling subscriptions and events to a pub sub adapter object.
+Current pubsub adapter classes are:
+
+ - in-memory event emitter based class which comes bundled in the subscription manager and is used mostly for development stages.
+ - Redis PubSub adapter which manages the publish and subscribe on a redis server - https://github.com/davidyaha/graphql-redis-subscriptions
+ - Postgres LISTEN/NOTIFY and update triggers as a pubsub adapter - https://github.com/jtmthf/graphql-postgres-subscriptions/
+ - Mongo oplog reader that publishes it’s messages to channel names based on collection name and action taken - https://github.com/sebakerckhof/graphql-mongo-subscriptions/
+
+*There is no way to use more than one adapter (Other than creating a custom mixed adapter)
+The server has no standard way to figure out which user/session he is publishing data to.*
+
+# Aggregation / throttling of events
+
+GraphQL main goal is user experience for devices inside poor networks.
+
+therefore giving the client ability to configure how frequent the updates will be is very helpful. (poor network -> lower updates, standard-network -> regular updates)
```

Comment:
Actually i meant as resolver arguments, but ofcourse it can apply to directive also, i liked your idea